$NetBSD: patch-configure,v 1.4 2026/09/14 22:10:12 tnn Exp $

pcap_version is not portable; use pcap_lib_version(), which returns a
string like "libpcap version 1.10.7 (...)", so skip to the first digit
before parsing it.

External liblinear must be configured later.

--- configure.orig	2025-05-06 17:33:27.000000000 +0000
+++ configure
@@ -6274,9 +6274,12 @@ else
 /* end confdefs.h.  */
 
 #include <stdio.h>
-extern char pcap_version[];
+#include <pcap/pcap.h>
 int main() {
-  int major, minor1, minor2;
+  int major = 0, minor1 = 0, minor2 = 0;
+  const char *pcap_version = pcap_lib_version();
+  while (*pcap_version != '\0' && (*pcap_version < '0' || *pcap_version > '9'))
+    pcap_version++;
   sscanf(pcap_version,"%d.%d.%d", &major, &minor1, &minor2);
   if (major > 0)
     return 0;
@@ -6310,9 +6313,12 @@ else
 #ifdef HAVE_NET_BPF_H
 #include <net/bpf.h>
 #endif
-extern char pcap_version[];
+#include <pcap/pcap.h>
 int main() {
-  int major, minor;
+  int major = 0, minor = 0;
+  const char *pcap_version = pcap_lib_version();
+  while (*pcap_version != '\0' && (*pcap_version < '0' || *pcap_version > '9'))
+    pcap_version++;
   sscanf(pcap_version,"%d.%d", &major, &minor);
   if ((major == 1 && minor >= 1) || major > 1)
     return 0;
@@ -7196,7 +7202,6 @@ if test "${with_liblinear+set}" = set; t
   *)
     CPPFLAGS="-I$with_liblinear/include $CPPFLAGS"
     LDFLAGS="-L$with_liblinear/lib $LDFLAGS"
-    have_liblinear=yes
   ;;
   esac
 
